### ISSUE TITLE
Update GH actions to publish to PyPI

### ISSUE THREAD

**Issue by javiermtorres** [mozilla-ai/lumigator#417]
**Is your feature request related to a problem? Please describe.**
Currently SDK and schemas packages are not published to PyPI.

**Describe the solution you'd like**
The GH action https://github.com/mozilla-ai/lumigator/blob/main/.github/workflows/build_and_push_sdk_packages.yaml should post to PyPI instead of Test PyPI.

**Describe alternatives you've considered**
N/A

**Additional context**
The versioning mechanism must be in place before posting to PyPI.

**Comment by macaab26** [comment]
I modified the existent actions to be executed by the creation of a tag, and added more conditionals in order to decide which tags we use in each scenario: tag, merge to main, create a PR, etc. TL,DR: When we create a tag, we tag the BE and FE images with that value, and we create a package for the SDK.

**Comment by macaab26** [comment]
Next Monday we will move from test-PyPI to PyPI (with a new PR).
